[BUG] Gradle Package Munching not working

This is a new Bug report following #224 and #225.

**Describe the bug**
We are receiving reports that as of plugin version `1.2.2` which includes #225, the plugin is unable to always obtain dependencies for projects that are using Gradle.

**To Reproduce**
Steps to reproduce the behaviour:
1. Clone https://github.com/sonatype-nexus-community/nexusiq-successmetrics
2. Use the VS Code Plugin configured with Nexus Lifecycle to analyse
3. See 

**Screenshots**
<img width="456" alt="Screenshot 2022-01-14 at 14 42 38" src="https://user-images.githubusercontent.com/10280392/149533821-adb036f6-e80d-4e5e-a1fa-0d3ada513271.png">


**Desktop (please complete the following information):**
 - VS Code Version: 1.63.2
 - Version of Extension: 1.2.2
 - System Gradle Version: 6.9.1

cc @bhamail / @DarthHater / @rpanman-sonatype / @guillermo-varela / @maurycupitt

@guillermo-varela  - here's what I get:

```
./gradlew dependencies --configuration runtimeClasspath


> Task :dependencies

------------------------------------------------------------
Root project 'successmetrics'
------------------------------------------------------------

runtimeClasspath - Runtime classpath of source set 'main'.
+--- org.springframework.boot:spring-boot-devtools -> 2.5.6
|    +--- org.springframework.boot:spring-boot:2.5.6
|    |    +--- org.springframework:spring-core:5.3.12
|    |    |    \--- org.springframework:spring-jcl:5.3.12
|    |    \--- org.springframework:spring-context:5.3.12
|    |         +--- org.springframework:spring-aop:5.3.12
|    |         |    +--- org.springframework:spring-beans:5.3.12
|    |         |    |    \--- org.springframework:spring-core:5.3.12 (*)
|    |         |    \--- org.springframework:spring-core:5.3.12 (*)
|    |         +--- org.springframework:spring-beans:5.3.12 (*)
|    |         +--- org.springframework:spring-core:5.3.12 (*)
|    |         \--- org.springframework:spring-expression:5.3.12
|    |              \--- org.springframework:spring-core:5.3.12 (*)
|    \--- org.springframework.boot:spring-boot-autoconfigure:2.5.6
|         \--- org.springframework.boot:spring-boot:2.5.6 (*)
+--- org.springframework.boot:spring-boot-starter -> 2.5.6
|    +--- org.springframework.boot:spring-boot:2.5.6 (*)
|    +--- org.springframework.boot:spring-boot-autoconfigure:2.5.6 (*)
|    +--- org.springframework.boot:spring-boot-starter-logging:2.5.6
|    |    +--- ch.qos.logback:logback-classic:1.2.6
|    |    |    +--- ch.qos.logback:logback-core:1.2.6
|    |    |    \--- org.slf4j:slf4j-api:1.7.32
|    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.14.1
|    |    |    +--- org.slf4j:slf4j-api:1.7.25 -> 1.7.32
|    |    |    \--- org.apache.logging.log4j:log4j-api:2.14.1
|    |    \--- org.slf4j:jul-to-slf4j:1.7.32
|    |         \--- org.slf4j:slf4j-api:1.7.32
|    +--- jakarta.annotation:jakarta.annotation-api:1.3.5
|    +--- org.springframework:spring-core:5.3.12 (*)
|    \--- org.yaml:snakeyaml:1.28
+--- org.springframework.boot:spring-boot-starter-actuator -> 2.5.6
|    +--- org.springframework.boot:spring-boot-starter:2.5.6 (*)
|    +--- org.springframework.boot:spring-boot-actuator-autoconfigure:2.5.6
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.12.5
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.12.5
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.12.5
|    |    |    |         +--- junit:junit:4.13.1 -> 4.13.2 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-annotations:2.12.5 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-core:2.12.5 (c)
|    |    |    |         +--- com.fasterxml.jackson.core:jackson-databind:2.12.5 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.12.5 (c)
|    |    |    |         +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.5 (c)
|    |    |    |         \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.12.5 (c)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.12.5
|    |    |    |    \--- com.fasterxml.jackson:jackson-bom:2.12.5 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.12.5 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.5
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.12.5 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.12.5 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.12.5 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.12.5 (*)
|    |    +--- org.springframework.boot:spring-boot-actuator:2.5.6
|    |    |    \--- org.springframework.boot:spring-boot:2.5.6 (*)
|    |    +--- org.springframework.boot:spring-boot:2.5.6 (*)
|    |    \--- org.springframework.boot:spring-boot-autoconfigure:2.5.6 (*)
|    \--- io.micrometer:micrometer-core:1.7.5
|         +--- org.hdrhistogram:HdrHistogram:2.1.12
|         \--- org.latencyutils:LatencyUtils:2.0.3
+--- org.springframework.boot:spring-boot-starter-thymeleaf -> 2.5.6
|    +--- org.springframework.boot:spring-boot-starter:2.5.6 (*)
|    +--- org.thymeleaf:thymeleaf-spring5:3.0.12.RELEASE
|    |    +--- org.thymeleaf:thymeleaf:3.0.12.RELEASE
|    |    |    +--- ognl:ognl:3.1.26
|    |    |    |    \--- org.javassist:javassist:3.20.0-GA -> 3.27.0-GA
|    |    |    +--- org.attoparser:attoparser:2.0.5.RELEASE
|    |    |    +--- org.unbescape:unbescape:1.1.6.RELEASE
|    |    |    \--- org.slf4j:slf4j-api:1.7.25 -> 1.7.32
|    |    \--- org.slf4j:slf4j-api:1.7.25 -> 1.7.32
|    \--- org.thymeleaf.extras:thymeleaf-extras-java8time:3.0.4.RELEASE
|         +--- org.thymeleaf:thymeleaf:3.0.11.RELEASE -> 3.0.12.RELEASE (*)
|         \--- org.slf4j:slf4j-api:1.7.25 -> 1.7.32
+--- org.springframework.boot:spring-boot-starter-web -> 2.5.6
|    +--- org.springframework.boot:spring-boot-starter:2.5.6 (*)
|    +--- org.springframework.boot:spring-boot-starter-json:2.5.6
|    |    +--- org.springframework.boot:spring-boot-starter:2.5.6 (*)
|    |    +--- org.springframework:spring-web:5.3.12
|    |    |    +--- org.springframework:spring-beans:5.3.12 (*)
|    |    |    \--- org.springframework:spring-core:5.3.12 (*)
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.12.5 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.12.5
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.12.5 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.12.5 (*)
|    |    |    \--- com.fasterxml.jackson:jackson-bom:2.12.5 (*)
|    |    +--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.5 (*)
|    |    \--- com.fasterxml.jackson.module:jackson-module-parameter-names:2.12.5
|    |         +--- com.fasterxml.jackson.core:jackson-core:2.12.5 (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.12.5 (*)
|    |         \--- com.fasterxml.jackson:jackson-bom:2.12.5 (*)
|    +--- org.springframework.boot:spring-boot-starter-tomcat:2.5.6
|    |    +--- jakarta.annotation:jakarta.annotation-api:1.3.5
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:9.0.54
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:9.0.54
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:9.0.54
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:9.0.54
|    +--- org.springframework:spring-web:5.3.12 (*)
|    \--- org.springframework:spring-webmvc:5.3.12
|         +--- org.springframework:spring-aop:5.3.12 (*)
|         +--- org.springframework:spring-beans:5.3.12 (*)
|         +--- org.springframework:spring-context:5.3.12 (*)
|         +--- org.springframework:spring-core:5.3.12 (*)
|         +--- org.springframework:spring-expression:5.3.12 (*)
|         \--- org.springframework:spring-web:5.3.12 (*)
+--- org.springframework.boot:spring-boot-starter-webflux -> 2.5.6
|    +--- org.springframework.boot:spring-boot-starter:2.5.6 (*)
|    +--- org.springframework.boot:spring-boot-starter-json:2.5.6 (*)
|    +--- org.springframework.boot:spring-boot-starter-reactor-netty:2.5.6
|    |    \--- io.projectreactor.netty:reactor-netty-http:1.0.12
|    |         +--- io.netty:netty-codec-http:4.1.69.Final
|    |         |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    +--- io.netty:netty-buffer:4.1.69.Final
|    |         |    |    \--- io.netty:netty-common:4.1.69.Final
|    |         |    +--- io.netty:netty-transport:4.1.69.Final
|    |         |    |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    |    \--- io.netty:netty-resolver:4.1.69.Final
|    |         |    |         \--- io.netty:netty-common:4.1.69.Final
|    |         |    +--- io.netty:netty-codec:4.1.69.Final
|    |         |    |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    |    \--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |    \--- io.netty:netty-handler:4.1.69.Final
|    |         |         +--- io.netty:netty-common:4.1.69.Final
|    |         |         +--- io.netty:netty-resolver:4.1.69.Final (*)
|    |         |         +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |         +--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |         \--- io.netty:netty-codec:4.1.69.Final (*)
|    |         +--- io.netty:netty-codec-http2:4.1.69.Final
|    |         |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-codec:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-handler:4.1.69.Final (*)
|    |         |    \--- io.netty:netty-codec-http:4.1.69.Final (*)
|    |         +--- io.netty:netty-resolver-dns:4.1.69.Final
|    |         |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-resolver:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-codec:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-codec-dns:4.1.69.Final
|    |         |    |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    |    +--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |    |    \--- io.netty:netty-codec:4.1.69.Final (*)
|    |         |    \--- io.netty:netty-handler:4.1.69.Final (*)
|    |         +--- io.netty:netty-resolver-dns-native-macos:4.1.69.Final
|    |         |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    +--- io.netty:netty-resolver-dns:4.1.69.Final (*)
|    |         |    \--- io.netty:netty-transport-native-unix-common:4.1.69.Final
|    |         |         +--- io.netty:netty-common:4.1.69.Final
|    |         |         +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |         \--- io.netty:netty-transport:4.1.69.Final (*)
|    |         +--- io.netty:netty-transport-native-epoll:4.1.69.Final
|    |         |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |    \--- io.netty:netty-transport-native-unix-common:4.1.69.Final (*)
|    |         +--- io.projectreactor.netty:reactor-netty-core:1.0.12
|    |         |    +--- io.netty:netty-handler:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-handler-proxy:4.1.69.Final
|    |         |    |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    |    +--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |    |    +--- io.netty:netty-codec:4.1.69.Final (*)
|    |         |    |    +--- io.netty:netty-codec-socks:4.1.69.Final
|    |         |    |    |    +--- io.netty:netty-common:4.1.69.Final
|    |         |    |    |    +--- io.netty:netty-buffer:4.1.69.Final (*)
|    |         |    |    |    +--- io.netty:netty-transport:4.1.69.Final (*)
|    |         |    |    |    \--- io.netty:netty-codec:4.1.69.Final (*)
|    |         |    |    \--- io.netty:netty-codec-http:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-resolver-dns:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-resolver-dns-native-macos:4.1.69.Final (*)
|    |         |    +--- io.netty:netty-transport-native-epoll:4.1.69.Final (*)
|    |         |    \--- io.projectreactor:reactor-core:3.4.11
|    |         |         \--- org.reactivestreams:reactive-streams:1.0.3
|    |         \--- io.projectreactor:reactor-core:3.4.11 (*)
|    +--- org.springframework:spring-web:5.3.12 (*)
|    \--- org.springframework:spring-webflux:5.3.12
|         +--- org.springframework:spring-beans:5.3.12 (*)
|         +--- org.springframework:spring-core:5.3.12 (*)
|         +--- org.springframework:spring-web:5.3.12 (*)
|         \--- io.projectreactor:reactor-core:3.4.11 (*)
+--- org.springframework.boot:spring-boot-starter-data-jpa:2.5.3
|    +--- org.springframework.boot:spring-boot-starter-aop:2.5.3 -> 2.5.6
|    |    +--- org.springframework.boot:spring-boot-starter:2.5.6 (*)
|    |    +--- org.springframework:spring-aop:5.3.12 (*)
|    |    \--- org.aspectj:aspectjweaver:1.9.7
|    +--- org.springframework.boot:spring-boot-starter-jdbc:2.5.3 -> 2.5.6
|    |    +--- org.springframework.boot:spring-boot-starter:2.5.6 (*)
|    |    +--- com.zaxxer:HikariCP:4.0.3
|    |    |    \--- org.slf4j:slf4j-api:1.7.30 -> 1.7.32
|    |    \--- org.springframework:spring-jdbc:5.3.12
|    |         +--- org.springframework:spring-beans:5.3.12 (*)
|    |         +--- org.springframework:spring-core:5.3.12 (*)
|    |         \--- org.springframework:spring-tx:5.3.12
|    |              +--- org.springframework:spring-beans:5.3.12 (*)
|    |              \--- org.springframework:spring-core:5.3.12 (*)
|    +--- jakarta.transaction:jakarta.transaction-api:1.3.3
|    +--- jakarta.persistence:jakarta.persistence-api:2.2.3
|    +--- org.hibernate:hibernate-core:5.4.32.Final
|    |    +--- org.jboss.logging:jboss-logging:3.4.1.Final -> 3.4.2.Final
|    |    +--- org.javassist:javassist:3.27.0-GA
|    |    +--- net.bytebuddy:byte-buddy:1.10.22
|    |    +--- antlr:antlr:2.7.7
|    |    +--- org.jboss:jandex:2.2.3.Final
|    |    +--- com.fasterxml:classmate:1.5.1
|    |    +--- org.dom4j:dom4j:2.1.3
|    |    +--- org.hibernate.common:hibernate-commons-annotations:5.1.2.Final
|    |    |    \--- org.jboss.logging:jboss-logging:3.3.2.Final -> 3.4.2.Final
|    |    \--- org.glassfish.jaxb:jaxb-runtime:2.3.1 -> 2.3.5
|    |         +--- jakarta.xml.bind:jakarta.xml.bind-api:2.3.3
|    |         +--- org.glassfish.jaxb:txw2:2.3.5
|    |         +--- com.sun.istack:istack-commons-runtime:3.0.12
|    |         \--- com.sun.activation:jakarta.activation:1.2.2
|    +--- org.springframework.data:spring-data-jpa:2.5.3 -> 2.5.6
|    |    +--- org.springframework.data:spring-data-commons:2.5.6
|    |    |    +--- org.springframework:spring-core:5.3.11 -> 5.3.12 (*)
|    |    |    +--- org.springframework:spring-beans:5.3.11 -> 5.3.12 (*)
|    |    |    \--- org.slf4j:slf4j-api:1.7.26 -> 1.7.32
|    |    +--- org.springframework:spring-orm:5.3.11 -> 5.3.12
|    |    |    +--- org.springframework:spring-beans:5.3.12 (*)
|    |    |    +--- org.springframework:spring-core:5.3.12 (*)
|    |    |    +--- org.springframework:spring-jdbc:5.3.12 (*)
|    |    |    \--- org.springframework:spring-tx:5.3.12 (*)
|    |    +--- org.springframework:spring-context:5.3.11 -> 5.3.12 (*)
|    |    +--- org.springframework:spring-aop:5.3.11 -> 5.3.12 (*)
|    |    +--- org.springframework:spring-tx:5.3.11 -> 5.3.12 (*)
|    |    +--- org.springframework:spring-beans:5.3.11 -> 5.3.12 (*)
|    |    +--- org.springframework:spring-core:5.3.11 -> 5.3.12 (*)
|    |    \--- org.slf4j:slf4j-api:1.7.26 -> 1.7.32
|    \--- org.springframework:spring-aspects:5.3.9 -> 5.3.12
|         \--- org.aspectj:aspectjweaver:1.9.7
+--- com.h2database:h2:2.0.202
+--- org.webjars:jquery:3.5.1
+--- org.webjars:bootstrap:4.5.0
|    +--- org.webjars:jquery:3.5.1
|    \--- org.webjars:popper.js:1.16.0
+--- org.thymeleaf:thymeleaf:3.0.12.RELEASE (*)
+--- org.xhtmlrenderer:flying-saucer-pdf-openpdf:9.1.20
|    +--- com.github.librepdf:openpdf:1.3.11
|    |    \--- com.github.albfernandez:juniversalchardet:2.3.0
|    \--- org.xhtmlrenderer:flying-saucer-core:9.1.20
+--- org.apache.commons:commons-csv:1.5
+--- com.opencsv:opencsv:5.2
|    +--- org.apache.commons:commons-lang3:3.10 -> 3.12.0
|    +--- org.apache.commons:commons-text:1.8
|    |    \--- org.apache.commons:commons-lang3:3.9 -> 3.12.0
|    +--- commons-beanutils:commons-beanutils:1.9.4
|    |    +--- commons-logging:commons-logging:1.2
|    |    \--- commons-collections:commons-collections:3.2.2
|    \--- org.apache.commons:commons-collections4:4.4
+--- org.apache.httpcomponents:httpclient:4.5.13
|    +--- org.apache.httpcomponents:httpcore:4.4.13 -> 4.4.14
|    +--- commons-logging:commons-logging:1.2
|    \--- commons-codec:commons-codec:1.11 -> 1.10
+--- commons-codec:commons-codec:1.10
+--- commons-io:commons-io:2.11.0
+--- org.json:json:20210307
\--- com.googlecode.json-simple:json-simple:1.1.1
     \--- junit:junit:4.10 -> 4.13.2
          \--- org.hamcrest:hamcrest-core:1.3 -> 2.2
               \--- org.hamcrest:hamcrest:2.2

(c) - dependency constraint
(*) - dependencies omitted (listed previously)

A web-based, searchable dependency report is available by adding the --scan option.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
1 actionable task: 1 executed
```

```
gradle dependencies --configuration runtimeClasspath

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
'org.gradle.api.tasks.SourceSetContainer org.gradle.api.plugins.JavaPluginExtension.getSourceSets()'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
```

--- a/README.md
+++ b/README.md
@@ -20,6 +20,8 @@ Good news for polyglots -- the extension allows you to view violations across mu
 - Conan (any conan formatted `*.lock` files)
 - Golang (`dep` or `go mod`)
 - Gradle (`build.gradle` and gradle required) *Supports development dependency exclusion*
+  - Note: `./gradlew` or `./gradlew.bat` will take precedence over system `gradle`
+  - Note: Android projects are not currently supported
 - Maven (`pom.xml` and maven required) *Supports development dependency exclusion*
 - NPM/Yarn (`npm install` or `yarn install` required) *Supports development dependency exclusion*
 - Python:

--- a/ext-src/models/ComponentEntry.ts
+++ b/ext-src/models/ComponentEntry.ts
@@ -43,7 +43,7 @@ export class ComponentEntry implements TreeableModel {
   }
 
   public toString(): string {
-    return `${this.application.name}: ${this.format}: ${this.name} @ ${this.version}`;
+    return `${this.format}: ${this.name} @ ${this.version}`;
   }
 
   public maxPolicy(): number {

--- a/ext-src/packages/gradle/GradleUtils.ts
+++ b/ext-src/packages/gradle/GradleUtils.ts
@@ -14,10 +14,12 @@
  * limitations under the License.
  */
 import commandExists from "command-exists";
+import * as path from "path";
 import { Application } from "../../models/Application";
 import { exec } from "../../utils/exec";
 import { ILogger, LogLevel } from "../../utils/Logger";
 import { MavenPackage } from "../maven/MavenPackage";
+import { PackageDependenciesHelper } from "../PackageDependenciesHelper";
 import { PackageDependenciesOptions } from "../PackageDependenciesOptions";
 
 export class GradleUtils {
@@ -40,19 +42,25 @@ export class GradleUtils {
     let gradleCommand: string = "";
 
     try {
-      const gradleExists = commandExists.sync(this.GRADLE);
-      this.logger.log(LogLevel.DEBUG, `Does gradle exist as a command?`, gradleExists);
+      let gradewBatExists = PackageDependenciesHelper.doesPathExist(application.workspaceFolder, this.GRADLEW_BAT)
+      this.logger.log(LogLevel.DEBUG, `Does ${application.workspaceFolder} / ${this.GRADLEW_BAT} exist? ${gradewBatExists}`)
 
-      if (gradleExists) {
-        gradleCommandBaseCommand = this.GRADLE;
-
-        this.logger.log(LogLevel.INFO, `Set gradle Command Base Command`, gradleCommandBaseCommand);
+      // Favour any gradlew in the project over gradle on the PATH
+      if (process.platform === 'win32' && PackageDependenciesHelper.doesPathExist(application.workspaceFolder, this.GRADLEW_BAT)) {
+        this.logger.log(LogLevel.INFO, `Using ${this.GRADLEW_BAT}`)
+        gradleCommandBaseCommand = path.join(application.workspaceFolder, this.GRADLEW_BAT)
+      } else if (PackageDependenciesHelper.doesPathExist(application.workspaceFolder, this.GRADLEW)) {
+        this.logger.log(LogLevel.INFO, `Using ${this.GRADLEW}`)
+        gradleCommandBaseCommand = path.join(application.workspaceFolder, this.GRADLEW)
+      } else if (commandExists.sync(this.GRADLE)) {
+        this.logger.log(LogLevel.INFO, `Using ${this.GRADLE}`)
+        gradleCommandBaseCommand = this.GRADLE
       } else {
-        this.logger.log(LogLevel.INFO, `Operating system determination`, process.platform);
-
-        gradleCommandBaseCommand = (process.platform === 'win32') ? this.GRADLEW_BAT : this.GRADLEW;
-
-        this.logger.log(LogLevel.INFO, `Set gradle Command Base Command`, gradleCommandBaseCommand);
+        return Promise.reject(
+          new Error(
+            `Could not find a Gradle executable. Please check that ${gradleCommandBaseCommand} exists or is on your PATH.`
+          )
+        );
       }
 
       if (includeDev) {
@@ -62,7 +70,6 @@ export class GradleUtils {
       }
 
       this.logger.log(LogLevel.INFO, `Full gradle command constructed`, gradleCommand);
-
 
       this.logger.log(LogLevel.DEBUG, `Attempting to run gradle command`, gradleCommand);
       const { stdout, stderr } = await exec(gradleCommand, {
@@ -118,7 +125,7 @@ export class GradleUtils {
         let version: string = coords[2];
 
         if (artifact && group && version) {
-          if (version.includes("(*")) {
+          if (version.includes("(*") || version.includes("(c")) {
             console.warn("Omitted version");
             return;
           }

--- a/package.json
+++ b/package.json
@@ -203,7 +203,7 @@
   "dependencies": {
     "@agney/react-loading": "^0.1.2",
     "@fortawesome/free-solid-svg-icons": "^5.15.1",
-    "@sonatype/react-shared-components": "^9.0.1",
+    "@sonatype/react-shared-components": "^9.5.15",
     "@types/platform": "^1.3.2",
     "@types/temp": "^0.8.34",
     "command-exists": "^1.2.9",

